libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -1898,4 +1900,32 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
             routesPreviewController.previewNavigationRoutes(emptyList())
         }
     }
+
+    @Test
+    fun mapboxNavigationCreationSetsInstanceToHolder() {
+        mockkObject(LegacyMapboxNavigationInstanceHolder) {
+            createMapboxNavigation()
+            verify { LegacyMapboxNavigationInstanceHolder.onCreated(mapboxNavigation) }
+        }
+    }
+
+    @Test(expected = IllegalStateException::class)
+    fun mapboxNavigationIsNotCreatedIfHolderHasInstance() {
+        mockkObject(LegacyMapboxNavigationInstanceHolder) {
+            every { LegacyMapboxNavigationInstanceHolder.peek() } returns mockk(relaxed = true)
+            createMapboxNavigation()
+            verify(exactly = 0) {
+                LegacyMapboxNavigationInstanceHolder.onCreated(any())
+            }
+        }
+    }
+
+    @Test
+    fun mapboxNavigationDestructionRemovesInstanceFromHolder() {
+        mockkObject(LegacyMapboxNavigationInstanceHolder) {
+            createMapboxNavigation()
+            mapboxNavigation.onDestroy()
+            verify { LegacyMapboxNavigationInstanceHolder.onDestroyed() }
+        }
```

Comment:
What do you think is more important for us:
1. `MapboxNavigation` calls `LegacyMapboxNavigationInstanceHolder.onDestroyed()` during destroy
2. User can create a new instance of `MapboxNavigation` when the old one is destroyed 

?

Second statement seems more important to me, so I would test it instead
```suggestion
            createMapboxNavigation()
            mapboxNavigation.onDestroy()
            createMapboxNavigation()
```

Reply:
Your test would have passed even without the `LegacyMapboxNavigationInstanceHolder`...
What I want to test here is that `LegacyMapboxNavigationInstanceHolder` will be cleared. And what happens when it's cleared is tested in `LegacyMapboxNavigationInstanceHolderTest`.